docs/docs/dev_docs/contracts/syntax/historical_access/how_to_access_history.md

```diff
@@ -0,0 +1,99 @@
+---
+title: How to access history from a smart contract
+---
+
+The Aztec Protocol uses an append-only Merkle tree to store the headers of all previous blocks in the chain as its leaves. This is known as an archive tree. You can learn more about how it works in the [concepts section](../../../../concepts/advanced/data_structures/trees.md#archive-tree).
+
+# History library
+
+The history library allows you to access any of the following at a given block height before the current height:
```

Comment:
```suggestion
The history library allows you to prove any of the following at a given block height before the current height:
```

Technically you have to give the data to the library so it doesn't allow you to access it. It just allows you to prove the relevant statements if you provide it the relevant preimages.